eth_abi/registry.py

```diff
@@ -0,0 +1,146 @@
+from collections import namedtuple
+
+
+class ABIRegistryError(Exception):
+    pass
+
+
+class ExistingEntryFound(ABIRegistryError):
+    pass
+
+
+class NoEntryFound(ABIRegistryError):
+    pass
+
+
+PredicateItem = namedtuple('PredicateItem', ('predicate', 'value'))
+
+
+def ensure_changed(ExcClass, error_msg):
+    def decorator(old_method):
+        def new_method(self, *args, **kwargs):
```

Comment:
*naming convention nitpick*

We've got a somewhat consistent convention for decorators to use the names `fn`, `inner` and `outer`

```python
def simple_dec(fn):
    def inner(...):
         return fn(...)
    return inner

def args_dec(arg):
    def outer(fn):
         def inner(...):
             return fn(...)
        return inner
    return outer
```

I don't know if we have many in place that are method decorators, so no established convention for naming of the method fn.  I'd suggest something like `meth`, `method` or something else better I haven't though of.

Reply:
Sure, I can use that convention.  The naming convention I've been using is geared toward being as descriptive as possible but still reasonably concise which is hopefully helpful to people who are less strong with decorators.  I've seen names like `outer`, `inner` but I generally don't find them to be very meaningful.  But we can go with the way things have been done.

Reply:
I can get onboard with that.  +1 (no need to go update the other cases, we can just migrate our convention over time when that code gets touched).